gateway/README.md

```diff
@@ -2,116 +2,187 @@
+
+_NOTE: You don't actually need to expose API first, you can fetch token also while exposing service. After checking `Secure API` checkbox `Fetch token` button appears, you can then fetch token and just return without saving._
+
+#### Test the APIs with authentication
 
 ```bash
-kubectl create -f deployment.yaml,api-without-auth.yaml -n $KYMA_EXAMPLE_ENV
+# without token:
+curl -ik https://<hostname>.kyma.local
+# > 401 Origin authentication failed.
+
+# with token:
+curl -ik https://<hostname>.kyma.local -H 'Authorization: <token>'
+# > 200 Hello world
+```
+
+### Manual exposure through kubectl
+
+#### Additional prerequisites
+
+- [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) 1.10.0
+- Token fetched from Console UI (look at the NOTE in Fetch token section). Will be referenced later with **\<token\>**.
+- Namespace with istio injection enabled. Will be referenced later with **\<namespace\>**. You can enable istio injection in the namespace by labelling that namespace using:
+
+``` bash
+kubectl label namespace <namespace> istio-injection=enabled
+```
+
+#### Create service
+
+_NOTE: almost all steps in this tutorial are shown for lambda but you can apply them also to service deployment just by changing directory and in some places also names_
+
+Apply one of the deployment.yaml files from the lambda or service directory in this example.
+
+``` bash
+kubectl apply -f ./lambda/deployment.yaml -n <namespace>
+```
+
+#### Expose service without authentication
+
+``` bash
+kubectl apply -f ./lambda/api-without-auth.yaml -n <namespace>
+```
+
+##### How that works
+
+To expose service without authentication omit "authentication" field in the repository or set it to an empty list:
+
+```yaml
+authentication: []
+```
+
+You can also set:
+
+``` yaml
+authenticationEnabled: false
 ```
 
+what forces authentication to be disabled no matter what is in the "authentication" field.
+
+In this sample, both fields are omitted.
+
 #### Test the APIs without authentication
 
 ```bash
-# for the function:
-curl -i https://hello.kyma.local
-# for the service
-curl -ik https://http-db-service.ykyma.local/orders
+curl -ik https://hello.kyma.local
+# > 200 Hello world
 ```
 
-#### Test the APIs with authentication  
+#### Expose service with authentication
 
-Run the following commands in either the 'lambda' or the 'service' folder.
+``` bash
+# Create Api with default authentication settings
+kubectl apply -f ./lambda/api-with-default-auth.yaml -n <namespace>
 
-```bash
-kubectl apply -f api-with-auth.yaml -n $KYMA_EXAMPLE_ENV
+# OR
+
+# Create Api with custom authentication settings
+kubectl apply -f ./lambda/api-with-auth.yaml -n <namespace>
 ```
 
-Retrieve a token by accessing the [Dex Web application](https://dex-web.kyma.local/). Sign in with the admin@kyma.cx email address and the generic password from the [Dex ConfigMap](../../../resources/core/charts/dex/templates/pre-install-dex-config-map.yaml) file.
+##### How that works
```

Comment:
I am not sure if this section is really necessary given the comment below...

Examples should not give detailed explanations, but just illustrate a concept. It is already clear how it is done by reading the yaml files.

Reply:
I agree that examples aim is not to give detailed explanations, however, I created that explanation about "authentication" and "authenticationEnabled" fields because in my opinion the user may have problems after he starts editing these yamls. 

But if you say that reading the yaml gives enough knowledge, I agree that it should be deleted.